packages/solo/src/start.js

```diff
@@ -165,29 +166,19 @@ const buildSwingset = async (
     plugin: { ...plugin.endowments },
   };
 
-  const {
-    SOLO_OTEL_EXPORTER_PROMETHEUS_PORT,
-    SOLO_OTEL_RESOURCE_ATTRIBUTES,
-    ...restEnv
-  } = process.env;
-  const env = {
-    ...restEnv,
-    OTEL_EXPORTER_PROMETHEUS_PORT: SOLO_OTEL_EXPORTER_PROMETHEUS_PORT,
-    OTEL_RESOURCE_ATTRIBUTES: SOLO_OTEL_RESOURCE_ATTRIBUTES,
-  };
+  const env = Object.fromEntries(
+    Object.entries(process.env)
+      .filter(([k]) => !k.match(/^(OTEL_|SLOG)/)) // Remove chain controls.
+      .map(([k, v]) => [k.replace(/^SOLO_/, ''), v]), // Replace SOLO_ controls with chain version.
```

Comment:
Does that `replace` mean that `SOLO_LMDB_MAP_SIZE` and `LMDB_MAP_SIZE` would get mapped to the same thing, and one would randomly replace the other? I think we run both chain and solo from the same "start a test environment" command that their env vars should be disjoint but also tolerate each other's presence without confusion.

I'm wondering if this should start with a `match(/^SOLO_`, but I don't know what else `env` is used for in this app.

Reply:
Thanks for pointing this out.  I've now separated the `SOLO_`-prefixed variables and used them to override `env`.  `process.env` is used directly by the chain, so now at least it's predictable what will happen, and shouldn't cause conflicts (`SOLO_${var}` is used to override `${var}` only in the ag-solo).